homeassistant/components/timer/__init__.py

```diff
@@ -177,7 +177,7 @@ def state_attributes(self):
             newduration = duration
 
         event = EVENT_TIMER_STARTED
-        if self._state == STATUS_PAUSED:
+        if self._state == STATUS_ACTIVE or self._state == STATUS_PAUSED:
```

Comment:
This is not correct. It's not actually restarted if you start it at a different time ? 

Also test is missing.

Reply:
That's exactly the point, it is restarted.
Changing this behavior could break a lot automations.

If we really need not restartable timers, we could easily add those via optional parameter.

I added tests for the restart event.